core/data/generic/generic.go

```diff
@@ -0,0 +1,197 @@
+// Copyright (C) 2018 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package generic provides methods to test whether a type conforms to a generic
+// interface.
+package generic
+
+import (
+	"bytes"
+	"fmt"
+	"reflect"
+)
+
+type (
+	// T1 is a placeholder type that can be used for generic subsitutions.
+	T1 struct{}
+
+	// T2 is a placeholder type that can be used for generic subsitutions.
+	T2 struct{}
+
+	// T3 is a placeholder type that can be used for generic subsitutions.
+	T3 struct{}
+
+	// T4 is a placeholder type that can be used for generic subsitutions.
+	T4 struct{}
+
+	// Any is the type used for open types.
+	Any struct{}
+)
+
+var (
+	// T1Ty is the type of T1.
+	T1Ty = reflect.TypeOf(T1{})
+
+	// T2Ty is the type of T2.
+	T2Ty = reflect.TypeOf(T2{})
+
+	// T3Ty is the type of T3.
+	T3Ty = reflect.TypeOf(T3{})
+
+	// T4Ty is the type of T4.
+	T4Ty = reflect.TypeOf(T4{})
+
+	anyTy = reflect.TypeOf(Any{})
+)
+
+// Match is the result of Implements.
+type Match struct {
+	// Errors found matching the interface to the implementation.
+	Errors []error
+
+	// Bindings of generic type to matched type.
+	Bindings map[reflect.Type]reflect.Type
+}
+
+// Ok returns true if the type implemented the generic interface.
+func (m Match) Ok() bool { return len(m.Errors) == 0 }
+
+// Implements returns nil if ty implements the generic interface iface,
```

Comment:
this function does not return a pointer and thus will never return nil.

Reply:
Good spot on the stale comment. Will fix.